File: tests/bootstrap.php
<?php

namespace FixRefs\Tests;

// use FixRefs\Tests\MyFunctionTest;
// تحميل autoloader الخاص بـ Composer
require __DIR__ . '/../vendor/autoload.php';

// تحميل ملف include_files.php
require __DIR__ . '/../work.php';

require __DIR__ . '/../src/include_files.php';

use PHPUnit\Framework\TestCase;

class MyFunctionTest extends TestCase
{
    protected function assertEqualCompare(string $expected, string $input, string $result)
    {
        // --
        $result = preg_replace("/\r\n/", "\n", $result);
        $expected = preg_replace("/\r\n/", "\n", $expected);
        // --
        if ($result === $input && $result !== $expected) {
            $this->fail("No changes were made! The function returned the input unchanged:\n$result");
        } else {
            $this->assertEquals($expected, $result, "Unexpected result:\n$result");
        }
    }
}

File: python/tests/bootstrap.py
"""
Test bootstrap module

Base test class for all tests
"""

import os
import sys

# Add src to path
sys.path.insert(0, os.path.join(os.path.dirname(__file__), '..'))


class MyFunctionTest:
    """Base test class with utility methods"""
    
    def assertEqualCompare(self, expected: str, input_text: str, result: str):
        """
        Compare expected and result, fail if result equals input but not expected
        
        Args:
            expected: Expected output
            input_text: Original input
            result: Actual result
        """
        # Normalize line endings
        result = result.replace('\r\n', '\n')
        expected = expected.replace('\r\n', '\n')
        
        if result == input_text and result != expected:
            raise AssertionError(
                f"No changes were made! The function returned the input unchanged:\n{result}"
            )
        else:
            assert expected == result, f"Unexpected result:\n{result}"
